envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/groups/MetadataNodeGroup.kt

```diff
@@ -16,7 +16,10 @@ class MetadataNodeGroup(
     private val logger by logger()
 
     override fun hash(node: NodeV2): Group {
-        return createV2Group(node)
+        if (properties.supportV2Configuration) {
+            return createV2Group(node)
+        }
+        throw V2NotSupportedException(serviceName = null)
```

Comment:
Do I see correctly that it should never happen? V2 request should be rejected earlier by NodeMetadataValidator right? So this is only a safe-check?

Reply:
That's true. I'll remove it.